addon/lib/document.js

```diff
@@ -58,18 +70,29 @@ export default EmberObject.extend(Evented, {
   questionJexl: computed(function() {
     const questionJexl = new jexl.Jexl();
 
-    questionJexl.addTransform(
-      "answer",
-      slug =>
-        this.fields.find(field => field.question.slug === slug).answer.value
-    );
+    questionJexl.addTransform("answer", slug => this.findAnswer(slug));
 
     return questionJexl;
   }),
 
+  findAnswer(slug) {
+    const result = this.findField(slug);
+    return result && result.answer.value;
+  },
+
+  findField(slug) {
+    // Here it would be possible to extend the search range
+    // over the entire tree by calling findFieldInTree()
+    return this.fields.find(field => field.question.slug === slug);
+  },
+
   fields: computed(() => []).readOnly(),
```

Comment:
Exactly, this would throw if you try something like `this.get("fields", "foo")`. Actually we should mark all our computed properties as readonly, except those which we _do_ want to call `set` on - which is rarely needed.